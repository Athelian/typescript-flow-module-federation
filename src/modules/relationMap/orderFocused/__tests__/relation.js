import generateRelation from '../relation';

describe('collapse view', () => {
  it('no orderItem', () => {
    const order = {
      id: '1',
      orderItems: [],
    };
    const result = generateRelation(order, { isCollapsed: true });
    expect(result).toMatchSnapshot();
  });

  it('have orderItem', () => {
    const order = {
      id: '1',
      orderItems: [{ id: '1' }],
    };
    const result = generateRelation(order, { isCollapsed: true });
    expect(result).toMatchSnapshot();
  });
});

describe('normal view', () => {
  it('normal data', () => {
    const order = {
      id: '1',
      orderItems: [
        {
          id: '1',
          batches: [{ id: '1' }, { id: '2' }],
        },
        {
          id: '2',
          batches: [],
        },
      ],
    };
    const result = generateRelation(order, { isCollapsed: false });
    expect(result).toMatchSnapshot();
  });
});
