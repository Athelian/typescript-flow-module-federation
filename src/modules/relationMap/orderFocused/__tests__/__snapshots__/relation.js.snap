// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`collapse view have orderItem 1`] = `
Array [
  Object {
    "id": "1",
    "type": "ORDER_HEADER",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "1",
    "type": "ORDER",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "1",
    ],
    "type": "LINK-1-ORDER",
  },
  Object {
    "id": "1",
    "type": "ORDER_ITEM_ALL",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "1",
    ],
    "type": "LINK-1-ORDER",
  },
  Object {
    "id": "1",
    "type": "BATCH_ALL",
  },
]
`;

exports[`collapse view no orderItem 1`] = `
Array [
  Object {
    "id": "1",
    "type": "ORDER_HEADER",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "1",
    "type": "ORDER",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
]
`;

exports[`normal view normal data 1`] = `
Array [
  Object {
    "id": "1",
    "type": "ORDER_HEADER",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "1",
    "type": "ORDER",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "1",
    ],
    "type": "LINK-1-ORDER",
  },
  Object {
    "id": "1",
    "type": "ORDER_ITEM_ALL",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "1",
    ],
    "type": "LINK-1-ORDER",
  },
  Object {
    "id": "1",
    "type": "BATCH_ALL",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "2",
      "1",
    ],
    "type": "LINK-4-ORDER_ITEM",
  },
  Object {
    "id": "1",
    "type": "ORDER_ITEM",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "2",
      "1",
    ],
    "type": "LINK-1-BATCH",
  },
  Object {
    "id": "1",
    "type": "BATCH",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "1",
    "relatedIds": Array [
      "2",
    ],
    "type": "LINK-2-ORDER_ITEM",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "2",
    "relatedIds": Array [
      "2",
    ],
    "type": "LINK-4-BATCH",
  },
  Object {
    "id": "2",
    "type": "BATCH",
  },
  Object {
    "type": "",
  },
  Object {
    "id": "2",
    "relatedIds": Array [
      "2",
    ],
    "type": "LINK-4-ORDER_ITEM",
  },
  Object {
    "id": "2",
    "type": "ORDER_ITEM",
  },
  Object {
    "type": "",
  },
  Object {
    "type": "",
  },
]
`;
